src/OpenLoco/Windows/TileInspector.cpp

```diff
@@ -0,0 +1,425 @@
+        window->colours[0] = skin->colour_0B;
+        window->colours[1] = skin->colour_0C;
+
+        return window;
+    }
+
+    static void prepareDraw(window* self)
+    {
+        if (Input::isToolActive(WindowType::tileInspector))
+            self->activated_widgets |= (1 << widx::select);
+        else
+            self->activated_widgets &= ~(1 << widx::select);
+    }
+
+    static void draw(Ui::window* const self, Gfx::drawpixelinfo_t* const context)
+    {
+        // Draw widgets.
+        self->draw(context);
+
+        // Coord X/Y labels
+        {
+            auto args = FormatArguments::common(StringIds::tile_inspector_x_coord);
+            auto& widget = self->widgets[widx::xPos];
+            Gfx::drawString_494B3F(*context, self->x + widget.left - 15, self->y + widget.top + 1, Colour::black, StringIds::wcolour2_stringid, &args);
+        }
+        {
+            auto args = FormatArguments::common(StringIds::tile_inspector_y_coord);
+            auto& widget = self->widgets[widx::yPos];
+            Gfx::drawString_494B3F(*context, self->x + widget.left - 15, self->y + widget.top + 1, Colour::black, StringIds::wcolour2_stringid, &args);
+        }
+
+        // Coord X/Y values
+        {
+            FormatArguments args = {};
+            args.push<int16_t>(_currentPosition.x / OpenLoco::Map::tile_size);
+            auto& widget = self->widgets[widx::xPos];
+            Gfx::drawString_494B3F(*context, self->x + widget.left + 2, self->y + widget.top + 1, Colour::black, StringIds::tile_inspector_coord, &args);
+        }
+        {
+            FormatArguments args = {};
+            args.push<int16_t>(_currentPosition.y / OpenLoco::Map::tile_size);
+            auto& widget = self->widgets[widx::yPos];
+            Gfx::drawString_494B3F(*context, self->x + widget.left + 2, self->y + widget.top + 1, Colour::black, StringIds::tile_inspector_coord, &args);
+        }
+    }
+
+    static string_id getElementTypeName(const tile_element& element)
+    {
+        static const std::map<element_type, string_id> typeToString = {
+            { element_type::surface, StringIds::tile_inspector_element_type_surface },
+            { element_type::track, StringIds::tile_inspector_element_type_track },
+            { element_type::station, StringIds::tile_inspector_element_type_station },
+            { element_type::signal, StringIds::tile_inspector_element_type_signal },
+            { element_type::building, StringIds::tile_inspector_element_type_building },
+            { element_type::tree, StringIds::tile_inspector_element_type_tree },
+            { element_type::wall, StringIds::tile_inspector_element_type_wall },
+            { element_type::road, StringIds::tile_inspector_element_type_road },
+            { element_type::industry, StringIds::tile_inspector_element_type_industry },
+        };
+
+        return typeToString.at(element.type());
+    }
+
+    static string_id getObjectName(const tile_element& element)
```

Comment:
I really don't like this tedious mess of a function. All object types appear to have the name in the first position. Wouldn't it be better to just have them inherit the name field from a base 'object' struct?

Reply:
You could do:

```
template<typename T>
string_id getNameFromObject(T* obj)
{
    return obj->name;
}
```
If you don't want to make a base object.

Reply:
Hmm, trouble is, you still have to retrieve the object by its type via the manager, right? So we may not be saving that much code that way.

Reply:
Well you can go a level higher.
```
template< typename T>
string_id GetObjectName(uint16_t objectId){
    return ObjectManager::get<T>(objectId)->name;
}

Reply:
Yeah,that's what I was thinking as well. You'd only be saving about two lines of code that way, right? That doesn't seem worth it.

If we add functions to each of the tile elements that go straight to string id it'd be a different story, though. We could then invoke an `objectStringId` straight on the element in question. But to keep the header simple, those would have to go into Tile.cpp, not Tile.h. That might prevent the compiler from inlining them…? Again, not sure if it's worth it.